### PR TITLE
feat: APP-2733 - Implement Link component

### PR DESCRIPTION
**Pull request by thekidnamedkd**
## Description

-   Implements the `Link` component with variants
-   Storybook + tests

Task: [APP-2733](https://aragonassociation.atlassian.net/browse/APP-2733)

## Type of change

-   [x] New feature (non-breaking change which adds functionality)

## Checklist:

-   [x] I have selected the correct base branch.
-   [x] I have performed a self-review of my own code.
-   [x] I have commented my code, particularly in hard-to-understand areas.
-   [x] I have made corresponding changes to the documentation.
-   [x] My changes generate no new warnings.
-   [x] Any dependent changes have been merged and published in downstream modules.
-   [x] I ran all tests with success and extended them if necessary.
-   [x] I have updated the `CHANGELOG.md` file in the root folder of the package after the [UPCOMING] title and before
        the latest version.

[APP-2733]: https://aragonassociation.atlassian.net/browse/APP-2733?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/link/link.test.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        const linkElement = screen.getByTestId('link');
+        expect(linkElement).toHaveTextContent(label);
+    });
+
+    it('receives focus when tabbed to', () => {
+        render(<Link label="Clickable Link" href="http://example.com" />);
+        const linkElement = screen.getByTestId('link');
+        fireEvent.keyDown(linkElement, { key: 'Tab', code: 'Tab' });
+        expect(linkElement).toHaveClass('test-focus');
+    });
```
Comment (RuggeroCino) [inline src/components/link/link.test.tsx:79]: The `test-focus` class is always applied to the component, right? If we want to test that the component is focusable we should probably check that the `tabIndex` attribute is not negative.

File: src/components/link/link.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+};
+
+type Story = StoryObj<LinkStoryProps>;
+
+// can swap out the IconType for any icon in the iconList
+export const Default: Story = {
+    args: {
+        variant: 'primary',
+        iconRight: IconType.UPPER_RIGHT_ARROW,
+    },
```
Comment (RuggeroCino) [inline src/components/link/link.stories.tsx:61]: A "default" story right now is a story of a component with minimal configuration (only required attributes). Here we're applying all the args by default on all stories. I would suggest to keep these stories aligned as the one of the other components for now. We can then sync and decide how to structure these stories to showcase all the component variants

**Review by RuggeroCino — COMMENTED** [review]
Thanks @thekidnamedkd!

File: src/components/link/link.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        return (
+            <a
+                ref={ref}
+                onClick={!disabled ? onClick : undefined}
+                href={disabled ? undefined : href}
+                className={linkClassName}
+                {...(disabled && { tabIndex: -1, 'aria-disabled': 'true' })}
+                {...(external && !disabled && { target: '_blank', rel: 'noopener noreferrer' })}
+                {...props}
+                data-testid="link"
```
[the remark this answers is not in the document]
Comment (thekidnamedkd) [reply src/components/link/link.tsx:56]: even for disabled tests of the link?

File: src/components/link/link.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        const descriptionClassName = classNames('truncate', disabled ? disabledStyle : 'text-neutral-500');
+
+        return (
+            <a
+                ref={ref}
+                onClick={!disabled ? onClick : undefined}
+                href={disabled ? undefined : href}
+                className={linkClassName}
+                {...(disabled && { tabIndex: -1, 'aria-disabled': 'true' })}
+                {...(external && !disabled && { target: '_blank', rel: 'noopener noreferrer' })}
```
[the remark this answers is not in the document]
Comment (thekidnamedkd) [reply src/components/link/link.tsx:54]: apples and oranges. i think it's easy to forget target=_blank too. could leverage the api to require external with no default? where we wouldn't have that hint by default. also talked with selim about creating an uber primitive link as well which is just styled a tag and no trim.

[the remark this answers is not in the document]
Comment (thekidnamedkd) [reply src/components/link/link.tsx:54]: I feel like it also depends on how tightly coupled the external icon is with this implementation. Of course we have the option of other icons, but it seems like LINK_EXTERNAL is nearly 100% top of mind here.

**Review by RuggeroCino — COMMENTED** [review]
Fix we discussed during the call

**Review by RuggeroCino — APPROVED** [review]
Thanks for the implementation @thekidnamedkd! ✅

**Pull request merged**